src/main/java/it/gov/pagopa/rtd/ms/rtdmsdecrypter/service/BlobSplitterImpl.java

```diff
@@ -78,16 +78,23 @@ public Stream<BlobApplicationAware> split(BlobApplicationAware blob) {
           writeChunks(it, writer, blob);
           BlobApplicationAware tmpBlob = new BlobApplicationAware(
               blob.getBlobUri());
+          tmpBlob.setOriginalBlob(blob);
           tmpBlob.setOriginalBlobName(blob.getBlob());
           tmpBlob.setStatus(SPLIT);
           tmpBlob.setApp(blob.getApp());
           tmpBlob.setBlob(chunkName);
           tmpBlob.setBlobUri(
               blob.getBlobUri().substring(0, blob.getBlobUri().lastIndexOf("/")) + "/" + chunkName);
+          tmpBlob.setNumChunk(chunkNum);
           blobSplit.add(tmpBlob);
         }
+        
         chunkNum++;
       }
+      for (BlobApplicationAware blobApplicationAware : blobSplit) {
+        blobApplicationAware.setTotChunk(chunkNum+1);
```

Comment:
This will not update `chunkNum` and the `setTotChunk` will be always call with the same number. e.g. if `chunkNum` is 10 on each iterations will be 11 not more. Don't you agree?

Instead of foreach you can update it like:
```suggestion
        blobApplicationAware.setTotChunk(chunkNum + blobSplit.lenght); // or .size()
```